src/load.ts

```diff
@@ -0,0 +1,75 @@
+import has from './has';
+import Promise from './Promise';
+
+declare var require: Function;
+declare var define: any;
+
+export interface Load {
+	(module: ModuleObject, ...moduleIds: string[]): Promise<any[]>;
+}
+
+export interface ModuleObject {
+	id: string;
+}
+
+function getBaseId(moduleObject: ModuleObject): string {
+	var id: string = moduleObject.id;
+	var index: number = id.lastIndexOf('/');
+
+	if (index > -1) {
+		id = id.substring(0, index);
+	}
+	return id;
+}
+
+interface HostLoad {
+	(moduleIds: string[]): Promise<any[]>;
+}
+
+const hostLoad: HostLoad = (function () {
+	if (typeof define === 'function' && define.amd) {
+		return function (moduleIds: string[]): Promise<any[]> {
+			return new Promise(function (resolve) {
+				// TODO: Error path
+				require(moduleIds, function (...modules: any[]) {
+					resolve(modules);
+				});
+			});
+		};
+	}
+	else if (has('host-node')) {
+		return function (moduleIds: string[]): Promise<any[]> {
+			return new Promise(function (resolve, reject) {
+				try {
+					resolve(moduleIds.map(function (moduleId): any {
+						return require(moduleId);
+					}));
+				}
+				catch (error) {
+					reject(error);
+				}
+			});
+		};
+	}
+})();
+
+const load: Load = (function () {
+	if (!hostLoad) {
+		return function () {
+			return Promise.reject(new Error('Unknown loader'));
+		};
+	}
+	else {
+		return function (module: ModuleObject, ...moduleIds: string[]): Promise<any> {
+			var baseId = getBaseId(module);
```

Comment:
I'm not convinced this is going to work for Node modules...  It seems to me that Node module IDs report absolute paths and include the file extension.  Moreover, they also seem to include backslash rather than forward slash on Windows.

It makes no sense to me that Node's `module.id` reports the filename, but that seems to be what it does, and even [what it's documented to do](https://nodejs.org/api/modules.html#modules_module_id).